src/main/java/org/openrewrite/java/testing/junit5/MigrateFromHamcrest.java

```diff
@@ -0,0 +1,107 @@
+
+public class MigrateFromHamcrest extends Recipe {
+    @Override
+    public String getDisplayName() {
+        return "Migrate from Hamcrest Matchers to JUnit5";
+    }
+
+    @Override
+    public String getDescription() {
+        return "This recipe will migrate all Hamcrest Matchers to JUnit5 assertions.";
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new MigrationFromHamcrestVisitor();
+    }
+
+    private static class MigrationFromHamcrestVisitor extends JavaIsoVisitor<ExecutionContext> {
+
+        @Override
+        public J.MethodInvocation visitMethodInvocation(J.MethodInvocation method, ExecutionContext executionContext) {
+            J.MethodInvocation mi = super.visitMethodInvocation(method, executionContext);
+            MethodMatcher matcherAssertTrue = new MethodMatcher("org.hamcrest.MatchAssert assertThat(String, boolean)");
+            MethodMatcher matcherAssertMatcher = new MethodMatcher("org.hamcrest.MatcherAssert assertThat(*, org.hamcrest.Matcher)");
+            MethodMatcher matcherAssertMatcherWithReason = new MethodMatcher("org.hamcrest.MatcherAssert assertThat(String,*,org.hamcrest.Matcher)");
+
+            if (matcherAssertTrue.matches(mi)) {
+                //TODO simple
+            } else if (matcherAssertMatcher.matches(mi)) {
+                Expression hamcrestMatcher = mi.getArguments().get(1);
+                if (hamcrestMatcher instanceof J.MethodInvocation) {
+                    J.MethodInvocation matcherInvocation = (J.MethodInvocation)hamcrestMatcher;
+                    maybeRemoveImport("org.hamcrest.Matchers." + matcherInvocation.getSimpleName());
+                    maybeRemoveImport("org.hamcrest.MatcherAssert.assertThat");
+                    String targetAssertion = getTranslatedAssert(matcherInvocation);
+                    if (targetAssertion.equals("")) {
+                        return mi;
+                    }
+
+                    JavaTemplate template = JavaTemplate.builder(getTemplateForMatcher(matcherInvocation.getSimpleName()))
+                      .javaParser(JavaParser.fromJavaVersion().classpathFromResources(executionContext, "junit-jupiter-api-5.9"))
+                      .staticImports("org.junit.jupiter.api.Assertions." + targetAssertion)
+                      .build();
+
+                    Expression strippedMatcher = mi.getArguments().get(1);
+                    if (strippedMatcher instanceof J.MethodInvocation) {
+                        strippedMatcher = ((J.MethodInvocation) strippedMatcher).getArguments().get(0);
+                    } else {
+                        throw new IllegalArgumentException("Second parameter expected to be a matcher constructor call.");
+                    }
+                    template.apply(getCursor(), method.getCoordinates().replace(), mi.getArguments().get(0), strippedMatcher);
+                    maybeAddImport("org.junit.jupiter.api.Assertions", targetAssertion);
```

Comment:
Don't forget to return the changed template! With this return a first of your tests now passes 🎉 
```suggestion
                    maybeAddImport("org.junit.jupiter.api.Assertions", targetAssertion);
                    return template.apply(getCursor(), method.getCoordinates().replace(), mi.getArguments().get(0), strippedMatcher);
```